AGENTS.md

```diff
@@ -0,0 +1,187 @@
+# AGENTS.md
+
+This file provides guidance to AI code assistants when working with code in this repository.
+
+## Overview
+
+The Cluster Version Operator (CVO) is a core OpenShift Cluster Operator that manages cluster upgrades by reconciling release payload images to the cluster. It continuously monitors for available updates from the OpenShift Update Service (OSUS) and orchestrates the systematic rollout of new versions across all cluster operators.
```

Comment:
Do we want to unify this overview with the README overview?  It's probably worth trying to take the higher-level overview there too, and then only get down into the details of things like resource manifests in other places (like [here][1]).

[1]: https://github.com/openshift/enhancements/blob/master/dev-guide/cluster-version-operator/user/reconciliation.md#release-image-content